pkg/pillar/cmd/zedagent/handleappInstMetadata.go

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2021. Zededa, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package zedagent
+
+import "github.com/lf-edge/eve/pkg/pillar/types"
+
+func getAllAppInstMetaData(ctx *zedagentContext) map[string]*types.AppInstMetaData {
+	sub := ctx.subAppInstMetaData
+	appInstIDAndAppInstMetadata := make(map[string]*types.AppInstMetaData)
+	for appInstID, appInstMetaDataInf := range sub.GetAll() {
+		appInstMetaData := appInstMetaDataInf.(types.AppInstMetaData)
+		appInstIDAndAppInstMetadata[appInstID] = &appInstMetaData
+	}
+	return appInstIDAndAppInstMetadata
+}
+
+func handleAppInstMetaDataCreate(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	handleAppInstMetaDataImpl(ctxArg, key, statusArg)
+}
+
+func handleAppInstMetaDataModify(ctxArg interface{}, key string,
```

Comment:
Wouldn't it also make sense to have a delete handler?

What we do for other objects is to send an API invocation with just the UUID and no payload as an indication to the controller that the object has been deleted.

Reply:
Added.